src/ResourceManager/Sql/Commands.Sql/TransparentDataEncryption/Cmdlet/AzureRmSqlManagedInstanceKeyVaultKeyBase.cs

```diff
@@ -0,0 +1,173 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Common.Authentication.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.Server.Model;
+using Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Adapter;
+using Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Security;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+
+
+namespace Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Cmdlet
+{
+    /// <summary>
+    /// Defines the Add-AzureRmSqlManagedInstanceKeyVaultKey cmdlet
+    /// </summary>
+    public abstract class AzureRmSqlManagedInstanceKeyVaultKeyBase : AzureSqlCmdletBase<IEnumerable<AzureRmSqlManagedInstanceKeyVaultKeyModel>, AzureSqlDatabaseTransparentDataEncryptionArmAdapter>
+    {
+        /// <summary>
+        /// Parameter sets
+        /// </summary>
+        protected const string DefaultParameterSet = "AddAzureRmSqlManagedInstanceKeyVaultKeyDefaultParameterSet";
+        protected const string InputObjectParameterSet = "AddAzureRmSqlManagedInstanceKeyVaultKeyInputObjectParameterSet";
+        protected const string ResourceIdParameterSet = "AddAzureRmSqlManagedInstanceKeyVaultKeyResourceIdParameterSet";
+
+        /// <summary>
+        /// Gets or sets the Managed instance Object
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The managed instance input object")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlManagedInstanceModel ManagedInstance { get; set; }
```

Comment:
Please add an alias to InputObject here, and ResourceId below (all it requires is adding the attribute ```[Alias("InputObject")]``` to the parameter)

Reply:
Ping